include/nbl/builtin/hlsl/emulated/emulated_float64_t_utils.hlsl

```diff
@@ -0,0 +1,459 @@
+};
+
+}
+
+template<typename T, uint32_t N>
+using portable_vector_t = typename impl::portable_vector<T, N>::type;
+
+template<typename T>
+using portable_vector_t2 = portable_vector_t<T, 2>;
+template<typename T>
+using portable_vector_t3 = portable_vector_t<T, 3>;
+template<typename T>
+using portable_vector_t4 = portable_vector_t<T, 4>;
+
+using portable_vector64_t2 = portable_vector_t2<portable_float64_t<> >;
+using portable_vector64_t3 = portable_vector_t3<portable_float64_t<> >;
+using portable_vector64_t4 = portable_vector_t4<portable_float64_t<> >;
+
+template<typename T, uint32_t N, uint32_t M>
+using portable_matrix_t = typename impl::portable_matrix<T, N, M>::type;
+
+template<typename T>
+using portable_matrix_t2x2 = portable_matrix_t<T, 2, 2>;
+template<typename T>
+using portable_matrix_t3x3 = portable_matrix_t<T, 3, 3>;
+
+using portable_matrix64_t2x2 = portable_matrix_t2x2<portable_float64_t<> >;
+using portable_matrix64_t3x3 = portable_matrix_t3x3<portable_float64_t<> >;
+
+template<typename T>
+NBL_CONSTEXPR_INLINE_FUNC portable_float64_t<> create_portable_float64_t(T val)
+{
+    return _static_cast<portable_float64_t<> >(val);
+}
+
+template<typename T>
+NBL_CONSTEXPR_INLINE_FUNC portable_vector64_t2 create_portable_vector64_t2(T val)
+{
+    portable_vector64_t2 output;
+    output.x = create_portable_float64_t(val);
+    output.y = create_portable_float64_t(val);
+
+    return output;
+}
+
+template<typename X, typename Y>
+NBL_CONSTEXPR_INLINE_FUNC portable_vector64_t2 create_portable_vector64_t2(X x, Y y)
+{
+    portable_vector64_t2 output;
+    output.x = create_portable_float64_t(x);
+    output.y = create_portable_float64_t(y);
+
+    return output;
+}
+
+template<typename VecType>
+NBL_CONSTEXPR_INLINE_FUNC portable_vector64_t2 create_portable_vector64_t2_from_2d_vec(VecType vec)
+{
+    portable_vector64_t2 output;
+    output.x = create_portable_float64_t(vec.x);
+    output.y = create_portable_float64_t(vec.y);
+
+    return output;
+}
+
+template<typename T>
+NBL_CONSTEXPR_INLINE_FUNC portable_vector64_t3 create_portable_vector64_t3(T val)
+{
+    portable_vector64_t3 output;
+    output.x = create_portable_float64_t(val);
+    output.y = create_portable_float64_t(val);
+    output.z = create_portable_float64_t(val);
+
+    return output;
+}
+
+template<typename X, typename Y, typename Z>
+NBL_CONSTEXPR_INLINE_FUNC portable_vector64_t3 create_portable_vector64_t3(X x, Y y, Z z)
+{
+    portable_vector64_t3 output;
+    output.x = create_portable_float64_t(x);
+    output.y = create_portable_float64_t(y);
+    output.z = create_portable_float64_t(z);
+
+    return output;
+}
+
+template<typename VecType>
+NBL_CONSTEXPR_INLINE_FUNC portable_vector64_t3 create_portable_vector64_t2_from_3d_vec(VecType vec)
+{
+    portable_vector64_t3 output;
+    output.x = create_portable_float64_t(vec.x);
+    output.y = create_portable_float64_t(vec.y);
+    output.z = create_portable_float64_t(vec.z);
+
+    return output;
+}
```

Comment:
why are all these not partial specializations of `static_cast_helper` ?

Reply:
these functions are no more